scripts/write-config.sh

```diff
@@ -0,0 +1,206 @@
+#!/bin/bash
+#
+# write-config.sh extracts configuration information from vault and writes it to a set of files
+# in a directory. This simplifies access to the secrets from other scripts and applications.
+#
+# This is intended as a replacement for render_config.sh in the service project and the
+# render-config.sh in the integration project. We want to avoid writing into the source
+# tree. Instead, this will write into a subdirectory of the root directory.
+#
+# We want to use this in a gradle task, so it takes arguments both as command line
+# options and as envvars. For automations, like GHA, the command line can be specified.
+# For developer use, we can set our favorite envvars and let gradle ensure the directory is properly populated.
+#
+# The environment passed in is used to configure several other parameters, including the target for running
+# the integration tests.
+#
+# For personal environments, we assume that the target name is the same as the personal namespace name.
+# The output directory includes the following files:
+#   --------------------------+-------------------------------------------------------------------------
+#   buffer-client-sa.json     | Creds to access Buffer service
+#   --------------------------+-------------------------------------------------------------------------
+#   db-connection-name.txt    | Connection string for CloudSQL proxy
+#   --------------------------+-------------------------------------------------------------------------
+#   db-name.txt               | Database name
```

Comment:
nit: to make it parallel with `stairway-db-name.txt`, you could make these `wm-db-name.txt`